internal/k8s/controller_test.go

```diff
@@ -85,6 +85,55 @@ func TestIsNginxIngress(t *testing.T) {
 		},
 	}
 
+	var testsWithoutIngressClassOnlyVS = []struct {
```

Comment:
this test grew 2 times now :) maybe it makes sense split VS/VSRs and Ingress into multiple test functions?